src/ducks/files/Toolbar.jsx

```diff
@@ -57,12 +63,22 @@ const mapStateToProps = (state, ownProps) => ({
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
   uploadFiles: (files, displayedFolder) => {
-    dispatch(uploadFiles(files, displayedFolder))
-      .catch(err => {
-        if (err.response && err.response.status === 413) {
-          alert(<QuotaAlert t={ownProps.t} />)
-        }
-      })
+    dispatch(addToUploadQueue(files, displayedFolder._id, file => uploadedFile(file), (loaded, quotas, conflicts, errors) => {
+      let action = { type: '' }// dummy action, we only use it to trigger an alert notification
+
+      if (quotas.length > 0) {
+        // quota errors have their own modal instead of a notification
+        alert(<QuotaAlert t={ownProps.t} />)
+      } else if (conflicts.length > 0) {
+        action.alert = alertShow('upload.alert.success_conflicts', {smart_count: loaded.length, conflictNumber: conflicts.length}, ALERT_LEVEL_INFO)
+      } else if (errors.length > 0) {
+        action.alert = alertShow('upload.alert.errors', null, ALERT_LEVEL_ERROR)
+      } else {
+        action.alert = alertShow('upload.alert.success', {smart_count: loaded.length}, ALERT_LEVEL_SUCCESS)
+      }
+
+      return action
```

Comment:
@goldoraf : I moved the alerts out of the upload duck. It makes it less coupled which is good, but mostly there's the quota limit error that is handled differently, and I couldn't think of a clean way to include it in the duck.